core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -237,33 +242,27 @@ protected String toPackageName(Class<?> clazz) {
 
     protected boolean isExcludedPackageNamePatterns(Class<?> clazz) {
         String packageName = toPackageName(clazz);
-        for (Pattern pattern : excludedPackageNamePatterns) {
-            if (pattern.matcher(packageName).matches()) {
-                return true;
-            }
-        }
-        return false;
+        return excludedPackageNamePatterns.stream().anyMatch(pattern -> pattern.matcher(packageName).matches());
     }
 
     protected boolean isExcludedPackageNames(Class<?> clazz) {
-        String suffixedPackageName = toPackageName(clazz) + ".";
-        for (String excludedPackageName : excludedPackageNames) {
-            if (suffixedPackageName.startsWith(excludedPackageName)) {
+        String packageName = toPackageName(clazz);
+        List<String> packageParts = Arrays.asList(packageName.split("\\."));
+        for (int i = 0; i < packageParts.size(); i++) {
```

Comment:
Operations proportional to number of package parts of class being checked. Previously it was proportional to number of excluded packages in configuration.

Note that we still retain the behaviour of sub-packages also being excluded.